### DISCUSSION TITLE
Confusion Over iOS C++ Integration

### DISCUSSION THREAD

**Discussion by dabarych** [microsoft/cpp_client_telemetry#1062] [Q&A]
Apologies if the answer here is simple but I'm a bit stuck with regards to integrating the SDK with iOS. Once I've built the 1DS SDK using ```build-ios.sh clean release```, how do I wrap the C++ headers for use in an iOS XCode application for Swift? Do I need to import these headers as an iOS framework, create a C++ implementation like [EventSender.cpp](https://github.com/microsoft/cpp_client_telemetry/blob/master/examples/cpp/EventSender/EventSender.cpp), and wrap that in Obective-C? The existing [cocoa-app](https://github.com/microsoft/cpp_client_telemetry/tree/master/examples/objc/cocoa-app) example appears to build itself using CMake and CMakeLists.txt but I'm interested in building with XCode instead.

On Android, doing this is much clearer as building the 1DS SDK produces a .AAR file that can be imported as a module and built with Android Studio. How can I accomplish something similar using XCode?

**Comment by lalitb** [comment]
@eduardo-camacho  - Sorry to tag you here, just in case you have a suggestion on iOS integration.

**Reply by eduardo-camacho** [reply]
@lalitb, not sure how to interface this on swift, but Teams uses swift so they might have answer, you might want to check with them. We should provide swift wrapper in a near future.

**Reply by dabarych** [reply]
Thanks for the response @eduardo-camacho.

Would you happen to know of any Teams mobile contacts that I could reach out to? In addition, what kind of timeframe should we expect for a Swift wrapper?

**Reply by eduardo-camacho** [reply]
You might contact Nishchith C P. There is not ETA for Swift wrapper yet.

**Reply by dabarych** [reply]
Hi @eduardo-camacho I touched base with Nischith and she showed me that she's consuming 1DS through a pod called OneDsCppSdk.iOS. If I try consuming this pod (installing it anew), the pod installer recognizes that I'm trying to get something from the 1DS repo but then tells me that the specific repo doesn't exist. Is this pod still supported?

<img width="727" alt="Screen Shot 2022-10-10 at 8 06 48 PM" src="https://user-images.githubusercontent.com/86632330/195152922-27f74713-75f2-4ac1-b7a6-07c8ff3173df.png">

**Reply by lalitb** [reply]
@dabarych  = The pod is trying to fetch the `cpp_client_telemetry_modules` repo, which is private Microsoft repo. In case the pod is using the github account of user invoking the pod, please ensure that the user's github account has the right permission for the Microsoft github org. Else, there should be way for pod to disable cloning this private github repo using some env var, as the iOS application won't need the modules from this repo. Please check with Nischith if it is possible for pod to not clone this repo.
